src/arrayMethodSort.js

```diff
@@ -4,9 +4,41 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction = defoultSortFunction) {
+    for (let i = 0; i < this.length - 1; i++) {
+      let current = this[i];
+
+      for (let j = i + 1; j < this.length; j++) {
+        const next = this[j];
+
+        if (compareFunction(current, next) > 0) {
+          current = next;
+        }
```

Comment:
The current implementation of the swap logic is incorrect. The 'current' variable is being updated, but the actual array elements are not being swapped. You need to swap the elements of the array itself.